packages/file-download/src/internals/file-download.ts

```diff
@@ -0,0 +1,125 @@
+import type {
+    ErrorHandler,
+    GetUserDataFromRequest,
+    HttpCrawlerOptions,
+    InternalHttpCrawlingContext,
+    InternalHttpHook,
+    RequestHandler,
+    RouterRoutes,
+} from '@crawlee/http';
+import {
+    HttpCrawler,
+    Router,
+} from '@crawlee/http';
+import type { Dictionary } from '@crawlee/types';
+
+export type FileDownloadErrorHandler<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    > = ErrorHandler<FileDownloadCrawlingContext<UserData, JSONData>>;
+
+export interface FileDownloadCrawlerOptions<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    > extends HttpCrawlerOptions<FileDownloadCrawlingContext<UserData, JSONData>> {}
+
+export type FileDownloadHook<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    > = InternalHttpHook<FileDownloadCrawlingContext<UserData, JSONData>>;
+
+export interface FileDownloadCrawlingContext<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    > extends InternalHttpCrawlingContext<UserData, JSONData, FileDownload> {}
+
+export type FileDownloadRequestHandler<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    > = RequestHandler<FileDownloadCrawlingContext<UserData, JSONData>>;
+
+/**
+    Provides a framework for the parallel file download using plain HTTP requests. The URLs to download are fed either from a static list of URLs or can be added dynamically from another source.
```

Comment:
this is not a valid jsdoc, not sure if it would render properly in the docs? i mean i would expect the `*` on each line, as we do in all other places (also the indent is weird, why 5 spaces?)